src/main/java/viewmodel/admin/AdminOrderViewModel.java

```diff
@@ -0,0 +1,80 @@
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package viewmodel.admin;
+
+import business.customer.Customer;
+import business.order.CustomerOrder;
+import business.order.CustomerOrderDetails;
+import business.order.CustomerOrderLineItem;
+import business.product.Product;
+import viewmodel.BaseAdminViewModel;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.List;
+
+public class AdminOrderViewModel extends BaseAdminViewModel {
+
+    private String orderId;
+    private CustomerOrderDetails details;
+    private Customer customer;
+    private List<Product> products;
+    private CustomerOrder orderRecord;
+    private List<CustomerOrderLineItem> orderedProducts;
+
+    public AdminOrderViewModel(HttpServletRequest request) {
+        super(request);
+
+        this.orderId = request.getPathInfo().split("/")[1];
+        this.details = customerOrderService.getOrderDetails(Long.parseLong(orderId));
+        this.customer = details.getCustomer();
+        this.products = details.getProducts();
+        this.orderRecord = details.getCustomerOrder();
+        this.orderedProducts = details.getCustomerOrderLineItems();
+    }
+
+    public Customer getCustomer() {
+        return customer;
+    }
+
+    public List<Product> getProducts() {
+        return products;
+    }
+
+    public CustomerOrder getOrderRecord() {
+        return orderRecord;
+    }
+
+    public List<CustomerOrderLineItem> getOrderedProducts() {
+        return orderedProducts;
+    }
+
```

Comment:
Hello @nowucca 

One quick question:

This is from customers.jsp.

    _<c:forEach var="customer" items="${p.customerList}" varStatus="iter">

        <tr class="${((iter.index % 2) == 1) ? 'lightBlue' : 'white'} tableRow"
            onclick="document.location.href='customer/${customer.customerId}'">

                <%-- Below anchor tags are provided in case JavaScript is disabled --%>
            <td><a href="<c:url value="/admin/customer/${customer.customerId}"/>" class="noDecoration">${customer.customerId}</a></td>
            <td><a href="<c:url value="/admin/customer/${customer.customerId}"/>" class="noDecoration">${customer.name}</a></td>
            <td><a href="<c:url value="/admin/customer/${customer.customerId}"/>" class="noDecoration">${customer.email}</a></td>
            <td><a href="<c:url value="/admin/customer/${customer.customerId}"/>" class="noDecoration">${customer.phone}</a></td>
        </tr>

    </c:forEach>_

Inside the forEach loop, customer is a local variable, and so I can leave it as is.
Your confirmation is appreciated.

Thank you so much!!